### PR TITLE
feat: local and remote trigger Java common SDK testing

### PR DESCRIPTION
**Pull request by typotter**
🎟️ towards FF-3085
Ticket FF-3085

👯‍♂️ **Related PRs**
- [sdk-common-jvm](https://github.com/Eppo-exp/sdk-common-jdk/pull/72)
- #57
- #58
- #59 

### Motivation
Changes are often made to the [sdk-test-data](https://github.com/Eppo-exp/sdk-test-data) repository to capture new behaviours, bugs and edge cases. When these changes are pushed to `main`, the SDKs are cloned locally (locally to the github action running) and their respective tests are run. These tests are set up and run by copies of the SDK test workflows - see [sdk-test-data workflow](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows/test-sdks.yml). There are a number of limitations to this setup:

- Test steps are copied from the SDK’s respective workflows; changes to the SDK test workflows need to be replicated in sdk-test-data and this is not obvious to devs
- The SDK tests are not able to run against in-flight changes to `sdk-test-data`
- When new test-data is committed that breaks an SDK, that breakage is not surfaced in the SDK’s repository until some action triggers its testing workflow (on demand, on pull-request, etc.).

### Description of Changes
- add java-common testing to the group!

Repos are green!

External to this Change
🚀 - Each SDK's testing workflow is enhanced into a [reusable workflow](https://docs.github.com/en/actions/sharing-automations/reusing-workflows), exposing parameters for SDK branch and the sdk-test-data branch to use in testing.

### CODE REVIEW

File: .github/workflows/test-sdks-remote.yml
Comment (felipecsl) [inline .github/workflows/test-sdks-remote.yml:23]: why `tp/apibranch`? same on the other file below

File: .github/workflows/test-sdks-remote.yml
Comment (aarsilv) [inline .github/workflows/test-sdks-remote.yml:23]: I assume you'll drop the ref when https://github.com/Eppo-exp/sdk-common-jdk/pull/72 is merged

File: .github/workflows/test-sdks.yml
Comment (aarsilv) [inline .github/workflows/test-sdks.yml:17]: 💪

File: .github/workflows/test-sdks.yml
Comment (aarsilv) [inline .github/workflows/test-sdks.yml:20]: don't forget to update this too 😛

**Review by aarsilv — COMMENTED** [review]
Looks good--will approve once tp/apibranch is merged and then purged from this PR 😛

**Comment by typotter** [comment]
temp refs removed! ptal!

**Review by felipecsl — APPROVED** [review]

**Pull request merged**
